File: content/src/main/java/org/sandboxpowered/silica/minecraft/Colour.java
package org.sandboxpowered.silica.minecraft;

import java.util.Locale;

public enum Colour {
    WHITE,
    ORANGE,
    MAGENTA,
    LIGHT_BLUE,
    YELLOW,
    LIME,
    PINK,
    GRAY,
    LIGHT_GRAY,
    CYAN,
    PURPLE,
    BLUE,
    BROWN,
    GREEN,
    RED,
    BLACK;

    private final String name;

    Colour() {
        this.name = name().toLowerCase(Locale.ENGLISH);
    }

    Colour(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }
}

File: modules/vanilla/common/src/main/kotlin/org/sandboxpowered/silica/vanilla/util/Colour.kt
package org.sandboxpowered.silica.vanilla.util

import org.sandboxpowered.silica.api.util.StringSerializable

enum class Colour(override val asString: String) : StringSerializable {
    BLACK("black"),
    RED("red"),
    GREEN("green"),
    BROWN("brown"),
    BLUE("blue"),
    PURPLE("purple"),
    CYAN("cyan"),
    LIGHT_GRAY("light_gray"),
    GRAY("gray"),
    PINK("pink"),
    LIME("lime"),
    YELLOW("yellow"),
    LIGHT_BLUE("light_blue"),
    MAGENTA("magenta"),
    ORANGE("orange"),
    WHITE("white");

    companion object {
        val ALL = values()
        val NAMES = ALL.map { it.asString }
        val NAME_ARRAY = NAMES.toTypedArray()
    }
}